src/mintermization.cc

```diff
@@ -0,0 +1,287 @@
+            const BDD op1 = graph_to_bdd(graph.children[0]);
+            const BDD op2 = graph_to_bdd(graph.children[1]);
+            return op1 + op2;
+        } else if (node.operator_type == FormulaNode::NEG) {
+            assert(graph.children.size() == 1);
+            const BDD op1 = graph_to_bdd(graph.children[0]);
+            return !op1;
+        } else
+            assert(false);
+    }
+
+    assert(false);
+}
+
+void Mata::Mintermization::minterms_to_aut(Mata::IntermediateAut& res, const Mata::IntermediateAut& aut,
+                                           const std::vector<BDD>& minterms)
+{
+    for (const auto& trans : aut.transitions) {
+            // for each t=(q1,s,q2)
+        const auto &symbol_part = trans.second.children[0];
+
+        size_t symbol = 0;
+        assert(trans_to_bddvar.count(&symbol_part));
+        const BDD &bdd = trans_to_bddvar[&symbol_part];
+
+        for (const auto &minterm: minterms) {
+            // for each minterm x:
+            if (!((bdd * minterm).IsZero())) {
+                // if for symbol s of t is BDD_s < x
+                // add q1,x,q2 to transitions
+                IntermediateAut::parse_transition(res, {trans.first.raw, std::to_string(symbol),
+                                                        trans.second.children[1].node.raw});
+            }
+            symbol++;
+        }
+    }
+}
+
+void Mata::Mintermization::minterms_to_aut_afa(Mata::IntermediateAut& res, const Mata::IntermediateAut& aut,
+                                           const std::vector<BDD>& minterms)
+{
+    for (const auto& trans : aut.transitions) {
+        for (const auto& ds_pair : lhs_to_disjuncts_and_states[&trans.first]) {
+            // for each t=(q1,s,q2)
+            const auto disjunct = ds_pair.first;
+
+            assert(trans_to_bddvar.count(disjunct));
+            const BDD &bdd = trans_to_bddvar[disjunct];
+
+            size_t symbol = 0;
+            for (const auto &minterm: minterms) {
+                // for each minterm x:
+                if (!((bdd * minterm).IsZero())) {
+                    // if for symbol s of t is BDD_s < x
+                    // add q1,x,q2 to transitions
+                    const auto str_symbol = std::to_string(symbol);
+                    FormulaNode node_symbol(FormulaNode::OPERAND, str_symbol, str_symbol,
+                                            Mata::FormulaNode::OperandType::SYMBOL);
+                    assert(ds_pair.second != nullptr);
+                    res.add_transition(trans.first, node_symbol, *ds_pair.second);
+                }
+                symbol++;
+            }
+        }
+    }
+}
+
+Mata::IntermediateAut Mata::Mintermization::mintermize(const Mata::IntermediateAut& aut)
+{
+    if (!aut.is_nfa() && !aut.is_afa() && aut.alphabet_type == IntermediateAut::BITVECTOR) {
+        throw std::runtime_error("We currently support mintermization only for NFA with bitvectors");
```

Comment:
It should probably say that we support AFA too.

Reply:
done